math/binary_exponentiation.cpp

```diff
@@ -0,0 +1,43 @@
+#include<iostream>
+#include<cassert>
+
+// the concept is to do exponentiation by converting the power to a sum of powers of 2.
+// example: 3^13  we can convert 13 to 8+4+1. which can be represented by 1101. so to do 3^13 we can do instead
+// 3^8 x 3^4 x 3^1. and so we get 3^13 and since every exponent is square of the old exponent eg: 3^8=(3^4)^2.
+
+long long binary_exponentiation(ll a,ll b){
```

Comment:
Functions should be documented explaining the use of its parameters, a brief description, and what it returns.